wordpress-mobile/WordPress-Android
# Gb/js bundle in wpandroid directly

This PR re-introduces the packaging of the gutenberg-mobile JS bundle into wpandroid directly, to avoid complex inter-dependencies between Travis, jitpack and Travis again.

Changes in the PR:

1. Directly download the proper (based on the gutenberg-mobile submodule hash) JS bundle from S3 and package in in the assets as `index.android.bundle`
2. Disable the block editor in the UI, in preparation to merging to develop. There is a [separate PR](https://github.com/wordpress-mobile/WordPress-Android/pull/8695) that will introduce the proper UI to enable/disable the block editor.

### To test A (no Gutenberg)

1. Build wpandroid in any flavor with the `wp.BUILD_GUTENBERG_FROM_SOURCE` gradle flag set to false or not existing at all
2. Notice the download of the JS file from S3 (`Downloading JS bundle from https://s3-us-west-1.amazonaws.com/gutenberg-mobile-js-bundle/wordpress-mobile/gutenberg-mobile/2f024a4e835512dd73b60f9a4ea98081db388543/android/App.js`)
3. The app should build and run just fine, and no Gutenberg in sight

### To test B (Gutenberg forced)

1. Edit `EditPostActivity.java` and set `mShowGutenbergEditor` to true, and `mShowAztecEditor` and `mShowNewEditor` to false to force Gutenberg
2. Follow steps 1, 2 in Test A
3. Open a gutenberg post and notice the block editor working fine
